File: 07-steroid_langs/1-scala/src/main/scala/codely/contexts/rrss/posts/application/find/PostFinder.scala
package codely.contexts.rrss.posts.application.find

import codely.contexts.rrss.posts.domain.{PostDoesNotExistError, PostRepository, Post}

class PostFinder(repository: PostRepository) {
	def find(id: String): Either[PostDoesNotExistError, Post] = {
		repository.search(id) match {
			case None => Left(PostDoesNotExistError(id))
			case Some(post) => Right(post)
		}
	}
}

File: 08-langs_particularities/1-java/src/main/java/tv/codely/contexts/rrss/posts/application/find/PostFinder.java
package tv.codely.contexts.rrss.posts.application.find;

import tv.codely.contexts.rrss.posts.domain.Post;
import tv.codely.contexts.rrss.posts.domain.PostDoesNotExistError;
import tv.codely.contexts.rrss.posts.domain.PostRepository;

public record PostFinder(PostRepository repository) {
    public Post find(String id) throws PostDoesNotExistError {
        var post = repository.search(id);

        if (post == null) {
            throw new PostDoesNotExistError(id);
        }

        return post;
    }
}
